### ISSUE TITLE
UTM SE: iPadOS trackpad / mouse not working [Solved]

### ISSUE THREAD

**Issue by Ph0enixKM** [utmapp/UTM#6497]
**Describe the issue**
When using my iPad attached to the magic keyboard I cannot interact with emulated OS using my mouse or trackpad. This defeats the use of UTM SE for some serious work on an iPad.

**Configuration (required)**
* UTM Version: 4.5.3 (99)
* OS Version: iPadOS 17.5.1
* Device Model: iPad Pro M4
* Is it jailbroken (name jailbreak used)? No
* How did you install UTM? App Store

**Crash log**
N/A

**Debug log**
N/A

**Upload VM**
This issue does not seem to apply to a single virtual machine.

# Solution

Update to iPadOS 17.6

**Comment by isgallagher** [comment]
Also a problem on Apple Vision Pro with Magic Keyboard and Magic Trackpad.

**Comment by michael-leroy** [comment]
I have the same hardware as @Ph0enixKM and can replicate this issue with a Windows 98SE VM. An external USB mouse seems to work, but for the magic keyboard only left click seems to work.

**Comment by AlexTech01** [comment]
I have the same issue with both UTM (the JIT version) and UTM SE on the App Store, both the iPad Pro M4's built-in trackpad and a Bluetooth Magic Trackpad don't seem to work at all in the VM, but my Mac's trackpad when connected through Continuity does seem to work. I'm running an M4 iPad Pro on iOS 18.

**Comment by isgallagher** [comment]
@AlexTech01  How do you get the JIT version installed? I must have missed the documentation/tutorial on that.

**Comment by AlexTech01** [comment]
> @AlexTech01 How do you get the JIT version installed? I must have missed the documentation/tutorial on that.

I have a paid developer account and a Mac so I downloaded the UTM.ipa from the GitHub releases, made an app ID and provisioning profile in my account, granted it Extended Virtual Addressing and Increased Memory Limit entitlements, and then signed & installed it and used SideJITServer to enable JIT on every launch

**Comment by mirrornoir** [comment]
I can convey that I have heard multiple reports of this issue with M4 iPad Pro and Magic Keyboard on the UTM Discord.  If it helps, I have an M4 iPad Pro 11” with the Logitech Combo Touch keyboard case which does not experience this issue with either the keyboard or touchpad inputs. This issue seems to be limited exclusively to the brand new Magic Keyboard for M4 iPad Pros.  Users of earlier Magic Keyboards and iPads have reported no issues in response to these concerns. 

I confirmed my own results using iOS 17.5.1 using both UTM SE (App Store) and UTM (AltStore + SideJITServer).

**Comment by Added-Glue** [comment]
> I can convey that I have heard multiple reports of this issue with M4 iPad Pro and Magic Keyboard on the UTM Discord. If it helps, I have an M4 iPad Pro 11” with the Logitech Combo Touch keyboard case which does not experience this issue with either the keyboard or touchpad inputs. This issue seems to be limited exclusively to the brand new Magic Keyboard for M4 iPad Pros. Users of earlier Magic Keyboards and iPads have reported no issues in response to these concerns.
> 
> I confirmed my own results using iOS 17.5.1 using both UTM SE (App Store) and UTM (AltStore + SideJITServer).

I have the same setup with the Logitech Combo and have lost input from my trackpad while UTM SE is open (can't remember if the keyboard was still responding...). I think it might have had something to do with Apple's continuity and my MacBook, but was unable to find a fix without a reboot to UTM—it seems to work most of the time and occasionally not.

**Comment by Axenntio** [comment]
This issue is resolved now, the iOS 17.6 update resolve the trackpad issue.

**Comment by noracmiller** [comment]
I am still having this issue on iPadOS running the same specs as OP. 

UTM Version: 4.6.2
OS Version: iPadOS 18.1.1
Device Model: iPad Pro M4
Is it jailbroken (name jailbreak used)? No
How did you install UTM? App Store
Crash log
N/A

Debug log
N/A

Upload VM
It happens across VMs

**Comment by richardschwalb** [comment]
I’m running iPadOS 18.6 on an M4 13” iPad and the Magic Keyboard trackpad was working fine until this morning. Here’s what I did: 

- I started my Mac OS 9.2 VM and then went to change the screen resolution.
- Once the resolution changed it appeared the VM locked up as I couldn’t use the trackpad to control the cursor, so I used UTM’s controls to turn off the VM.
- I restarted the VM and noticed the trackpad was not controlling the cursor, but I can use finger gestures to control it.
- I shutdown the Mac OS 9.2 VM and started my WinXP VM thinking this was a Mac OS issue. Again, the trackpad does not control the cursor in WinXP.

So, whatever happened when I changed the Mac OS 9.2 screen resolution to kill trackpad support seems to affect all VMs.

Does anyone have a solution?
